testing/testing.go

```diff
@@ -99,3 +113,57 @@ func DivulgeAddresses(a, b network.Network) {
 	addrs := a.Peerstore().Addrs(id)
 	b.Peerstore().AddAddrs(id, addrs, peerstore.PermanentAddrTTL)
 }
+
+// MockConnectionGater is a mock connection gater to be used by the tests.
```

Comment:
This seems potentially useful as a test helper that could be re-used.
I'm not sure if I'd look for it here. Probably only after checking go-libp2p/p2p/net/mock and near go-libp2p-core/connmgr

Reply:
Hey @willscott 

I agree. Since this interface and PR is currently heavy WIP, I'd like to keep it here till it lands and then pull it out to /net/mock. Have added a TODO item for this to https://github.com/libp2p/go-libp2p/issues/872 to track it.